packages/graph-engine/src/nodes/color/matchAlpha.ts

```diff
@@ -0,0 +1,108 @@
+import { ColorSchema, NumberSchema } from '../../schemas/index.js';
+import { INodeDefinition } from '../../index.js';
+import { Node } from '../../programmatic/node.js';
+import { toColor } from './lib/utils.js';
+
+export default class NodeDefinition extends Node {
+	static title = 'Match Alpha';
+	static type = 'studio.tokens.color.matchAlpha';
+	static description =
+		'Finds the alpha value that, when used to blend the foreground and background colors, will result in the reference color.';
+	constructor(props: INodeDefinition) {
+		super(props);
+
+		this.addInput('foreground', {
+			type: {
+				...ColorSchema
+			}
+		});
+		this.addInput('background', {
+			type: {
+				...ColorSchema
+			}
+		});
+		this.addInput('reference', {
+			type: {
+				...ColorSchema
+			}
+		});
+		this.addInput('precision', {
+			type: {
+				...NumberSchema,
+				default: 0.01
+			}
+		});
+
+		this.addOutput('alpha', {
+			type: NumberSchema
+		});
+	}
+
+	execute(): void | Promise<void> {
+		const { foreground, background, reference, precision } =
+			this.getAllInputs();
+
+		const bg = toColor(background);
+		const fg = toColor(foreground);
+		const ref = toColor(reference);
+
+		let alpha = Number.NaN;
+
+		// the matching is done per channel
+
+		// if the background and reference are "the same" (within precision), return zero
+		if (
+			Math.abs(bg.r - ref.r) < precision &&
+			Math.abs(bg.g - ref.g) < precision &&
+			Math.abs(bg.b - ref.b) < precision
+		) {
+			alpha = 0;
+		} else {
+			// find the matching alpha for each channel
+			let ar = (ref.r - bg.r) / (fg.r - bg.r);
+			let ag = (ref.g - bg.g) / (fg.g - bg.g);
+			let ab = (ref.b - bg.b) / (fg.b - bg.b);
+
+			// set out-of-range alphas to NaN
+			if (ar < 0 || ar > 1) ar = Number.NaN;
```

Comment:
You can make this a single function
function normalize(value){
 if (value>=0 && ar<=1){
 return value
 }
 return Number.Nan;
} 

and then 

`const ar =normalize((ref.r - bg.r) / (fg.r - bg.r));`